pyartifactory/objects/repository.py

```diff
@@ -35,18 +37,32 @@ def get_repo(self, repo_name: str) -> AnyRepositoryResponse:
         """
         Finds repository in artifactory. Raises an exception if the repo doesn't exist.
         :param repo_name: Name of the repository to retrieve
-        :return: Either a local, virtual or remote repository
+        :return: Either a local, virtual, remote or federated repository
         """
         try:
             response = self._get(f"api/{self._uri}/{repo_name}")
+            response_data = response.json()
+            rclass = None
+
             try:
-                artifact_info: AnyRepositoryResponse = LocalRepositoryResponse.model_validate(response.json())
-            except ValidationError:
-                try:
-                    artifact_info = VirtualRepositoryResponse.model_validate(response.json())
-                except ValidationError:
-                    artifact_info = RemoteRepositoryResponse.model_validate(response.json())
-            return artifact_info
+                rclass = response_data["rclass"]
+            except KeyError:
+                raise KeyError('"rclass" key not found in the response data received by artifactory.')
+
+            # Match to the correct repository type depending on the rclass
+            if rclass == RClassEnum.local:
+                return LocalRepositoryResponse.model_validate(response_data)
+            elif rclass == RClassEnum.virtual:
+                return VirtualRepositoryResponse.model_validate(response_data)
+            elif rclass == RClassEnum.remote:
+                return RemoteRepositoryResponse.model_validate(response_data)
+            elif rclass == RClassEnum.federated:
+                return FederatedRepositoryResponse.model_validate(response_data)
+            else:
+                # this should never happen and is a missing repotype in the library
+                raise ArtifactoryError(
+                    f"Unknown repository type found in response: {rclass}. Please report this issue.",
+                )
```

Comment:
I rebuilt this section as another repo type requires another nested section of try/catch and this was hard to read/maintain.  Functionality should be the same.

Reply:
... except, that now there is a error thrown when rcall does not match instead of a magic "i don't know the rcall enum. I think its a remote repo"

Reply:
I like that new approach. It's way more maintainable 👍🏿